### ISSUE TITLE
Enforce safe handler in collateralJoin.sol

### ISSUE THREAD

**Issue by MrDeadCe11** [open-dollar/od-contracts#587]
when join is called in collateralJoin.sol:

```
  function join(address _account, uint256 _wad) external {
    safeEngine.transferInternalCoins(address(this), _account, RAY * _wad);
    systemCoin.transferFrom(msg.sender, address(this), _wad);
    systemCoin.burn(_wad);
    emit Join(msg.sender, _account, _wad);
  }
```
  
  the _account variable should always be a valid safehandler address, this should be enforced in the contract.

**Comment by pi0neerpat** [comment]
Risk is only to user's joining funds into the system without a safeHandler, and thus not having the ability to retrieve them

**Comment by pi0neerpat** [comment]
closed by #686
